src/formatDate.js

```diff
@@ -9,6 +9,67 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const partsOfOldDate = date.split(fromFormat[fromFormat.length - 1]);
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const currentDate = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        currentDate.yearYYYY = partsOfOldDate[i];
+        currentDate.yearYY = partsOfOldDate[i][2] + partsOfOldDate[i][3];
+        break;
+
+      case 'YY':
+        currentDate.yearYY = partsOfOldDate[i];
+
+        if (+partsOfOldDate[i] < 30) {
+          currentDate.yearYYYY = `20${partsOfOldDate[i]}`;
+        } else {
+          currentDate.yearYYYY = `19${partsOfOldDate[i]}`;
+        }
+        break;
+
+      case 'MM':
+        currentDate.month = partsOfOldDate[i];
+        break;
+
+      case 'DD':
+        currentDate.day = partsOfOldDate[i];
+        break;
+
+      default:
+        break;
+    }
```

Comment:
Consider using a more concise approach for assigning date parts to the `currentDate` object. Instead of using a switch statement, you could directly map the format strings to object properties. This would make the code shorter and more readable.